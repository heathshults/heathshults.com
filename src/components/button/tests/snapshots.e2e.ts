import { snapIt } from '../../../test';

const component = 'button';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('a default button', `<hs-button>Action</hs-button>`);
    snap('an a button', `<hs-button href="nowhere.com" value="Action"></hs-button>`);
    snap('a colored button', `<hs-button type="info">Action</hs-button>`);
    snap('a sized button', `<hs-button size="small">Action</hs-button>`);
    snap('a full-width button', `<hs-button full>Action</hs-button>`);
    snap('a ghost button', `<hs-button ghost>Action</hs-button>`);
    snap('a colored ghost button', `<hs-button ghost type="info">Action</hs-button>`);
    snap('a rounded button', `<hs-button rounded>Action</hs-button>`);
    snap('a rounded ghost button', `<hs-button rounded ghost>Action</hs-button>`);
    snap('an active button', `<hs-button active>Action</hs-button>`);
  });
});
