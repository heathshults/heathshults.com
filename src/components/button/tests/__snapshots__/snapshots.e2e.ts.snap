// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders a colored button 1`] = `"<hs-button type=\\"info\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--info\\">Action</button></hs-button>"`;

exports[`button renders a colored ghost button 1`] = `"<hs-button ghost=\\"\\" type=\\"info\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--ghost-info\\">Action</button></hs-button>"`;

exports[`button renders a default button 1`] = `"<hs-button class=\\"hydrated\\"><button class=\\"hs-button\\">Action</button></hs-button>"`;

exports[`button renders a full-width button 1`] = `"<hs-button full=\\"\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--block\\">Action</button></hs-button>"`;

exports[`button renders a ghost button 1`] = `"<hs-button ghost=\\"\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--ghost\\">Action</button></hs-button>"`;

exports[`button renders a rounded button 1`] = `"<hs-button rounded=\\"\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--rounded\\">Action</button></hs-button>"`;

exports[`button renders a rounded ghost button 1`] = `"<hs-button rounded=\\"\\" ghost=\\"\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--ghost hs-button--rounded\\">Action</button></hs-button>"`;

exports[`button renders a sized button 1`] = `"<hs-button size=\\"small\\" class=\\"hydrated\\"><button class=\\"hs-button u-small\\">Action</button></hs-button>"`;

exports[`button renders an a button 1`] = `"<hs-button href=\\"nowhere.com\\" value=\\"Action\\" class=\\"hydrated\\"><a href=\\"nowhere.com\\" class=\\"hs-button\\"></a></hs-button>"`;

exports[`button renders an active button 1`] = `"<hs-button active=\\"\\" class=\\"hydrated\\"><button class=\\"hs-button hs-button--active\\">Action</button></hs-button>"`;
